tests/fidelity.rs

```diff
@@ -72,8 +72,17 @@ fn test_fidelity() {
 
     thread::sleep(Duration::from_secs(6));
 
-    // TODO: Assert that fund request for fidelity is printed in the log.
-    // Add the logging functions and assert the log string.
+    // Assert that fund request for fidelity is printed in the log
+    let debug_log_path = "/tmp/coinswap/taker/debug.log";
+    if let Ok(log_contents) = std::fs::read_to_string(debug_log_path) {
+        assert!(
+            log_contents.contains("Send at least 0.01001000 BTC to"),
+            "Funding request message not found in debug.log"
+        );
+        log::info!("✅ Funding request message detected in logs");
+    } else {
+        panic!("Could not read debug.log file at {}", debug_log_path);
+    }
```

Comment:
Can we turn this into its own assertion function? 

We can do something like `assert_log("log string", <log_path>)` and use that in other tests where log assertion is needed.